### PR TITLE
v2: Switch from github.com/go-kit/kit to github.com/go-kit/log interfaces

### PR DESCRIPTION
**Pull request by liggitt**
github.com/go-kit/kit split out just the logging interfaces to github.com/go-kit/log to allow downstreams to avoid inheriting large numbers of additional dependencies (made visible and impacting to go modules in v0.10.0 of github.com/go-kit/kit). xref prometheus/common#255

This switches to the slimmed down logging interface module to drop a ton of transitive dependencies introduced by go-kit/kit v0.10.0

cc @dims

### CODE REVIEW

**Comment by liggitt** [comment]
cc @johanbrandhorst

File: README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ myServer := grpc.NewServer(
    * [`tags`](interceptors/tags) - a library that adds a `Tag` map to context, with data populated from request body
    * [`zap`](providers/zap) - integration of [zap](https://github.com/uber-go/zap) logging library into gRPC handlers.
    * [`logrus`](providers/logrus) - integration of [logrus](https://github.com/sirupsen/logrus) logging library into gRPC handlers.
-   * [`kit`](providers/kit) - integration of [go-kit](https://github.com/go-kit/kit/tree/master/log) logging library into gRPC handlers.
+   * [`kit`](providers/kit) - integration of [go-kit/log](https://github.com/go-kit/log/tree/main) logging library into gRPC handlers.
```
Comment (johanbrandhorst) [inline README.md:64]: Same comment here 😁 
```suggestion
   * [`kit`](providers/kit) - integration of [go-kit/log](https://github.com/go-kit/log) logging library into gRPC handlers.
```

**Review by johanbrandhorst — APPROVED** [review]

**Comment by liggitt** [comment]
done

**Comment by johanbrandhorst** [comment]
Thanks for your contribution!

**Pull request merged**